docs/api-reference/tasks-api.md

```diff
@@ -25,77 +25,1448 @@ sidebar_label: Tasks
+<code class="getAPI">GET</code> `/druid/indexer/v1/pendingTasks`
 
-Retrieve the status of a task.
+This endpoint retrieves an array of pending task objects in the current Druid cluster. It is functionally equivalent to `/druid/indexer/v1/tasks?state=pending`. It supports a set of optional query parameters to filter results. 
 
-`GET /druid/indexer/v1/task/{taskId}/segments`
+#### Query parameters
+|Parameter|Type|Description|
+|---|---|---|
+| `datasource`|String| Return tasks filtered by Druid datasource.|
+| `createdTimeInterval`|String (ISO-8601)| Return tasks created within the specified interval. |
+| `max`|Integer|Maximum number of `"complete"` tasks to return. Only applies when `state` is set to `"complete"`.|
+| `type`|String|Filter tasks by task type. See [task documentation](../ingestion/tasks.md) for more details.|
+
+#### Responses
+
+<!--DOCUSAURUS_CODE_TABS-->
+
+<!--200 SUCCESS-->
+<br/>
+*Successfully retrieved list of pending tasks* 
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+#### Sample request
+
+<!--DOCUSAURUS_CODE_TABS-->
+
+<!--cURL-->
+```shell
+curl "{domain}/druid/indexer/v1/pendingTasks"
+```
+<!--HTTP-->
+```HTTP
+GET /druid/indexer/v1/pendingTasks HTTP/1.1
+Host: {domain}
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+#### Sample response
+
+<details>
+  <summary>Click to show sample response</summary>
+  
+  ```json
+  [
+    {
+        "id": "query-7b37c315-50a0-4b68-aaa8-b1ef1f060e67",
+        "groupId": "query-7b37c315-50a0-4b68-aaa8-b1ef1f060e67",
+        "type": "query_controller",
+        "createdTime": "2023-06-23T19:53:06.037Z",
+        "queueInsertionTime": "2023-06-23T19:53:06.037Z",
+        "statusCode": "RUNNING",
+        "status": "RUNNING",
+        "runnerStatusCode": "PENDING",
+        "duration": -1,
+        "location": {
+            "host": null,
+            "port": -1,
+            "tlsPort": -1
+        },
+        "dataSource": "wikipedia_api",
+        "errorMsg": null
+    },
+    {
+        "id": "query-544f0c41-f81d-4504-b98b-f9ab8b36ef36",
+        "groupId": "query-544f0c41-f81d-4504-b98b-f9ab8b36ef36",
+        "type": "query_controller",
+        "createdTime": "2023-06-23T19:53:06.616Z",
+        "queueInsertionTime": "2023-06-23T19:53:06.616Z",
+        "statusCode": "RUNNING",
+        "status": "RUNNING",
+        "runnerStatusCode": "PENDING",
+        "duration": -1,
+        "location": {
+            "host": null,
+            "port": -1,
+            "tlsPort": -1
+        },
+        "dataSource": "wikipedia_api",
+        "errorMsg": null
+    }
+  ]
+  ```
+
+</details>
+
+### Get task payload
+
+#### URL
+<code class="getAPI">GET</code> `/druid/indexer/v1/task/{taskId}`
+
+Retrieves the payload of a task given the task id. It returns a JSON object with the task id and payload which includes task configuration details and relevant specifications associated with the execution of the task.
+
+#### Query parameters
+
+No query parameters.
```

Comment:
If an endpoint doesn't have query parameters, we can remove this section altogether.